icechunk/src/storage/s3.rs

```diff
@@ -119,7 +119,9 @@ pub async fn mk_client(config: &S3Options, credentials: S3Credentials) -> Client
 
     let mut s3_builder = Builder::from(&aws_config.load().await);
 
-    if config.allow_http {
+    if config.force_path_style.is_some_and(|x| x)
```

Comment:
I usually do `unwrap_or(false)` but see my previous comment on trying to make this an actual bool to match the rest